File: src/types/TreatmentSeries.ts
export type SeriesKind = 'quantity' | 'timer';

export interface TreatmentSeries {
  id: string;
  orderItemId: string;
  // API enrichment fields (present when loaded from real backend; absent in mock data)
  packageTypeName?: string;
  isTimerBased?: boolean;
  // Numeric progress fields (conditionally present depending on seriesKind)
  totalTreatments?: number;
  completedTreatments?: number;
  totalMinutes?: number;
  usedMinutes?: number;
  // Legacy fields preserved for mock-data and Phase < 010 compatibility
  packageTypeId?: string;
  customerId?: string;
  seriesKind?: SeriesKind;
  minutesPerTreatment?: number;
  createdDate?: string;
}

File: backend/BeautyCareClinic.Domain/Entities/TreatmentSeries.cs
namespace BeautyCareClinic.Domain.Entities;

public class TreatmentSeries
{
    public Guid Id { get; set; }
    public Guid OrderItemId { get; set; }
    public Guid CustomerId { get; set; }
    public int TotalTreatments { get; set; }
    public int CompletedTreatments { get; set; }
    public int TotalMinutes { get; set; }
    public int UsedMinutes { get; set; }

    public OrderItem OrderItem { get; set; } = null!;
    public Customer Customer { get; set; } = null!;
    public ICollection<Treatment> Treatments { get; set; } = new List<Treatment>();
}
